### PR TITLE
Segy Header Remapping

### PR DESCRIPTION
**Pull request by JesperDramsch**
### What does this PR do?
Allow header remapping for reading and writing.

### Why was it initiated?  Any relevant Issues?
#1775 

### PR Checklist
- [x] Correct base branch selected? `master` for new fetures, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [x] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+DOCS"
- [x] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:ALL"
      (you can also add "ALL" to just simply run all tests across all modules)
- [ ] All tests still pass.
- [ ] Any new features or fixed regressions are be covered via new tests.
- [ ] Any new or changed features have are fully documented.
- [ ] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

File: obspy/io/segy/segy.py [hunk 1]
```diff
@@ -116,6 +117,16 @@ def __init__(self, file=None, endian=None, textual_header_encoding=None,
         :param read_traces: Data traces will only be read if this is set to
             ``True``. The data will be completely ignored if this is set to
             ``False``.
+        :type map_header: list
+        :param non map_header: List of lists with headerwords that have been
```
Comment (heavelock) [inline obspy/io/segy/segy.py]: What's that 'non'? Is it marking not required param?

Comment (JesperDramsch) [reply obspy/io/segy/segy.py]: No it's a mistake on my side.

**Comment by JesperDramsch** [comment]
It did not seem like a priority and at this point I have basically forgotten what I did and how to fix the tests. So it's probably better if someone else has a go at it.

**Comment by megies** [comment]
OK, I understand. Sorry again that this didn't get sorted out. There's just too much going on always in obspy.. :-/

**Comment by JesperDramsch** [comment]
No worries, I just also didn't want to clog the pipeline here with a PR that is not particularly important.

**Comment by JesperDramsch** [comment]
No worries, I just also didn't want to clog the pipeline here with a PR that is not particularly important.

**Comment by JesperDramsch** [comment]
@megies Is circle CI supposed to fail? It's reporting errors on a bunch of things I didn't even change...

**Comment by megies** [comment]
@JesperDramsch that should be fixed if you rebase on current master again, see #2354

**Pull request closed without merging**